src/ResourceManager/ServerManagement/Commands.ServerManagement.Test/ScenarioTests/ServerManagementTests.ps1

```diff
@@ -0,0 +1,249 @@
+		Write-Verbose "Getting non-existant gateway by strings"
+		Assert-Throws { Get-AzureRmServerManagementGateway -resourceGroupName $resourceGroupName -gatewayname $GatewayName }
+		
+	} finally {
+	    # Remove all test gateways
+		Get-AzureRmServerManagementGateway  -resourceGroupName $resourceGroupName |? { $_.name -match "test_gateway_" } | Remove-AzureRmServerManagementGateway
+
+		# any left?
+		$gateways = (Get-AzureRmServerManagementGateway  -resourceGroupName $resourceGroupName |? { $_.name -match "test_gateway_" })
+		
+		if( $gateways ) { 
+			write-verbose "*sigh*"
+			# assert the fail manually, since the framework doesn't properly interpret empty results as $false
+			Assert-False { $true } "There should be no gateways called test_gateway_* still registered in the resource group $resourceGroupName" 
+		}
+	}
+}
+
+function Test-Node
+{
+    # resource group should exist
+	# make sure that we have a proper gateway functioning on this computer.
+	Ensure-GatewayIsCorrectlyConfigured
+
+	try {
+		$GatewayName = "mygateway"
+
+		Write-Verbose "Getting gateway."
+		$gateway = Get-AzureRmServerManagementGateway -ResourceGroupName $resourceGroupName -GatewayName $GatewayName -ea 0
+		
+		Write-Verbose "Verifying gateway is available."
+		assert-notnull $gateway
+
+		Write-Verbose "Creating an arbitrary node."
+		$node = $gateway | New-AzureRmServerManagementNode -NodeName "testnode" -Credential $creds
+		assert-notnull $node
+
+		Write-Verbose "Retrieving the arbitrary node."
+		$tmp = $node| Get-AzureRmServerManagementNode 
+		assert-notnull $tmp
+
+		Write-Verbose "Same node?"
+		assert-areequal $node.Name $tmp.Name
+
+	} finally {
+		# Remove all test nodes
+		$gateway  | Get-AzureRmServerManagementNode -ea 0 | Remove-AzureRmServerManagementNode
+		
+		$nodes = ($gateway | Get-AzureRmServerManagementNode -ea 0)
+
+		Assert-Null $nodes
+	}
+}
+
+function Test-Session
+{
+	# make sure that we have a proper gateway functioning on this computer.
+	Ensure-GatewayIsCorrectlyConfigured
+
+	try {
+		Create-NodeForThisPC
+
+		$node = GeT-AzureRmServerManagementNode -ResourceGroupName $resourceGroupName -NodeName $nodeName
+
+		Write-Verbose "Verifying we have a node."
+		Assert-NotNull $node
+
+		Write-Verbose "Creating session for $sessionId/$($node.Name)/$($node.GetType())"
+		$session = (New-AzureRmServerManagementSession $node -SessionName $sessionId -Credential $creds)
+		
+		Write-Verbose "Verifiyng session was created for $sessionId/$nodeUserName"
+		Assert-NotNull $session
+		Assert-AreEqual $nodeUserName $session.UserName 
+				
+		Write-Verbose "Invoke a powershell command."
+		$results = ($session | Invoke-AzureRmServerManagementPowerShellCommand -Command { dir c:\ })
+
+		Assert-NotNull $results
+		Write-Verbose "Results: `n$results"
+
+		Write-Verbose "Invoke a long-running powershell command."
+		$results = ($session | Invoke-AzureRmServerManagementPowerShellCommand -Command { dir c:\ ; sleep 20 ; dir 'C:\Program Files' })
```

Comment:
That code is executed on the remote server, it's not executed locally at all. (It's testing to see if long running command are handled ok).

That code doesn't actually get run in playback mode (since the result of the call is recorded)